testing/unittests/cryptotests.cpp

```diff
@@ -294,9 +301,100 @@ class DigiSignUnitTest : public CppUnit::TestFixture
+        {
+            Owned<CLoadedKey> publicKey = loadPublicKeyFromMemory(pubKey, nullptr);
+            Owned<CLoadedKey> privateKey = loadPrivateKeyFromMemory(privKey, nullptr);
+
+            // create random data
+            MemoryBuffer toEncryptMb;
+            fillRandomData(244, toEncryptMb); // max for RSA
+
+            MemoryBuffer pkeMb;
+            publicKeyEncrypt(pkeMb, toEncryptMb.length(), toEncryptMb.bytes(), *publicKey);
+
+            MemoryBuffer decryptedMb;
+            privateKeyDecrypt(decryptedMb, pkeMb.length(), pkeMb.bytes(), *privateKey);
+
+            ASSERT(toEncryptMb.length() == decryptedMb.length());
+            ASSERT(0 == memcmp(toEncryptMb.bytes(), decryptedMb.bytes(), toEncryptMb.length()));
+        }
+        catch (IException *e)
+        {
+            StringBuffer err;
+            e->errorMessage(err);
+            printf("pkeTests IException thrown:%s\n", err.str());
+            throw;
+        }
+        catch (CppUnit::Exception &e)
+        {
+            printf("pkeTests CppUnit::Exception thrown\n");
+            throw;
+        }
+    }
+
+    void aesWithRsaEncryptedKey()
+    {
+        try
+        {
+            // create random data
+            MemoryBuffer messageMb;
+            fillRandomData(1024*100, messageMb);
+
+            char aesKey[aesKeySize];
+            char aesIV[aesBlockSize];
+            fillRandomData(aesKeySize, aesKey);
+            fillRandomData(aesBlockSize, aesIV);
+
+            Owned<CLoadedKey> publicKey = loadPublicKeyFromMemory(pubKey, nullptr);
+
+            // Encrypt AES key with public key of server we're going to send to.
+            MemoryBuffer aesEncryptedKey;
+            publicKeyEncrypt(aesEncryptedKey, aesKeySize, aesKey, *publicKey);
+
+            MemoryBuffer encryptedMessageMb;
+            aesKeyEncrypt(encryptedMessageMb, messageMb.length(), messageMb.bytes(), aesKey, aesIV);
+
+
+            // Now imagine aesEncryptedKey + encryptedMessageMb + aesIV are transmitted to remote server
+
+
+            Owned<CLoadedKey> privateKey = loadPrivateKeyFromMemory(privKey, nullptr);
+
+            MemoryBuffer decryptedAesKeyMb;
+            privateKeyDecrypt(decryptedAesKeyMb, aesEncryptedKey.length(), aesEncryptedKey.bytes(), *privateKey);
+
+            ASSERT(decryptedAesKeyMb.length() == aesKeySize);
+            ASSERT(0 == memcmp(aesKey, decryptedAesKeyMb.bytes(), aesKeySize));
+
+            MemoryBuffer decryptedMessageMb;
+            aesKeyDecrypt(decryptedMessageMb, encryptedMessageMb.length(), encryptedMessageMb.bytes(), (const char *)decryptedAesKeyMb.bytes(), aesIV);
+
+            ASSERT(messageMb.length() == decryptedMessageMb.length());
+            ASSERT(0 == memcmp(messageMb.bytes(), decryptedMessageMb.bytes(), messageMb.length()));
+        }
+        catch (IException *e)
+        {
+            StringBuffer err;
+            e->errorMessage(err);
+            printf("pkeTests IException thrown:%s\n", err.str());
```

Comment:
Add the method name "aesWithRsaEncryptedKey"to the error string so you know if it failed here and not in pkeTests

Reply:
will fix.